lib/babe/build.go

```diff
@@ -322,7 +322,7 @@ func (b *BlockBuilder) addToQueue(txs []*transaction.ValidTransaction) {
 }
 
 func hasSlotEnded(slot Slot) bool {
-	slotEnd := slot.start.Add(slot.duration)
+	slotEnd := slot.start.Add(slot.duration * 2 / 3) // reserve last 1/3 of slot for block finalisation
```

Comment:
Instead of 1/3, Can we allocate a fixed time for block finalization?

Reply:
what do you mean exactly? 1/3 of the slot is a fixed slot duration, this is what substrate does afaik

Reply:
I meant `500ms` or so. Instead of `1/3` of `slot.duration`

Reply:
no that would not be ideal, substrate uses 1/3 of the slot duration, you can see their implementation on the issue